### PR TITLE
[jsarsasign]: update for v9 (breaking changes)

### PR DESCRIPTION
**Pull request by kratklu1**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## jsrsasign ([<kbd>unpkg</kbd>](https://unpkg.com/browse/jsrsasign@latest/))
was missing the following properties:
1. SecureRandom
1. rng_seed_time
1. ECDSA
1. DSA
1. Signature

<details>
<summary>as well as these 14 other properties...</summary>
<p>MessageDigest, Mac, Cipher, ASN1HEX, CryptoJS, hex2b64, ipv6tohex, hextoipv6, hextoip, iptohex, crypto, asn1, jws, lang</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 83f1e6e97f8c9b5b6b5f41f23977e1a94ac9be57
</p>

File: types/jsrsasign/modules/KJUR/asn1/csr/CSRUtil.d.ts [hunk 1]
```diff
@@ -103,5 +111,41 @@ declare namespace jsrsasign.KJUR.asn1.csr {
...
+         * get field values from CSR/PKCS#10 PEM string
+         * @param sPEM PEM string of CSR/PKCS#10
+         * @returns JSON object with parsed parameters such as name or public key
+         * @description
+         * This method parses PEM CSR/PKCS#1 string and retrieves
+         * fields such as subject name and public key.
+         * Following parameters are available in the
+         * resulted JSON object.
+         *
+         * - {X500Name}subject - subject name parameters
```
Comment (domdomegg) [inline types/jsrsasign/modules/KJUR/asn1/csr/CSRUtil.d.ts:125]: Might be worth moving this comment to the ParamResponse interface? But also appreciate may want to keep this together given it matches the docs.

File: types/jsrsasign/modules/KJUR/crypto/MessageDigest.d.ts [hunk 2]
```diff
@@ -91,7 +91,7 @@ declare namespace jsrsasign.KJUR.crypto {
          * @example
          * md.digest()
          */
-        digest(): void;
+        digest(): string;
```
Comment (domdomegg) [inline types/jsrsasign/modules/KJUR/crypto/MessageDigest.d.ts:94]: I think digestString and digestHex's return types should also be updated to string:

https://kjur.github.io/jsrsasign/wikistatic/Tutorial-for-MessageDigest-class.html
https://kjur.github.io/jsrsasign/api/symbols/KJUR.crypto.MessageDigest.html

File: types/jsrsasign/modules/KJUR/jws/JWS.d.ts [hunk 3]
```diff
@@ -234,7 +234,7 @@ declare namespace jsrsasign.KJUR.jws {
          */
         function verify(
             sJWS: string,
-            key: string,
+            key: string | RSAKey | crypto.DSA | crypto.ECDSA,
```
Comment (domdomegg) [inline types/jsrsasign/modules/KJUR/jws/JWS.d.ts:237]: I think this is right, but do you have the docs for it?

https://kjur.github.io/jsrsasign/api/symbols/KJUR.jws.JWS.html#.verify seems vague about how keys exactly are supported here.

File: types/jsrsasign/modules/X509.d.ts [hunk 4]
```diff
@@ -240,26 +278,29 @@ declare namespace jsrsasign {
...
+         * undefined - there is no basic constraints extension
          *
          * @example
          * x = new X509();
          * x.readCertPEM(sCertPEM); // parseExt() will also be called internally.
-         * x.getExtBasicConstraints() → { cA: true, pathLen: 3 };
+         * x.getExtBasicConstraints() &rarr; {cA:true,pathLen:3,critical:true}
          */
-        getExtBasicConstraints(): { cA: boolean; pathLen: number };
+        getExtBasicConstraints(): { cA: boolean; pathLen: number; critical?: boolean; extname?: string };
```
Comment (domdomegg) [inline types/jsrsasign/modules/X509.d.ts:303]:
```suggestion
        getExtBasicConstraints(hExtV?: string, critical?: boolean): { cA: boolean; pathLen: number; critical?: boolean; extname?: string };
```

**Review by domdomegg — CHANGES_REQUESTED** [review]
As per review comments

File: types/jsrsasign/modules/KJUR/crypto/MessageDigest.d.ts (hunk 2 above)
Comment (kratklu1) [reply types/jsrsasign/modules/KJUR/crypto/MessageDigest.d.ts:94]: Added

**Review by domdomegg — APPROVED** [review]
LGTM - thanks for the updated definitions 🙌

**Comment by ffflorian** [comment]
Sorry, I didn't have time to review it - thank you for your work!

**Comment by kratklu1** [comment]
Ready to merge

**Pull request merged**